crud/common/schema.lua

```diff
@@ -49,24 +48,29 @@ local function call_reload_schema(replicasets)
     return true
 end
 
-local reload_in_progress = false
-local reload_schema_cond = fiber.cond()
+local reload_in_progress = {}
+local reload_schema_cond = {}
 
-local function reload_schema(replicasets)
-    if reload_in_progress then
-        if not reload_schema_cond:wait(const.RELOAD_SCHEMA_TIMEOUT) then
+local function reload_schema(vshard_router)
+    local replicasets = vshard_router:routeall()
+
+    if reload_in_progress[vshard_router.name] == true then
+        if not reload_schema_cond[vshard_router.name]:wait(const.RELOAD_SCHEMA_TIMEOUT) then
```

Comment:
It is nice that you have noticed it. Since this behavior was here even before my patch, I filed an issue: https://github.com/tarantool/crud/issues/320

Reply:
Thank you!